linode/manager
# feat: [M3-7919] Add Fonoa tax id event notification on invalidation

## Description 📝
To enhance our tax ID validation process, we've updated Cloud Manager to offer clearer guidance on these UX changes.

## Changes  🔄
- Non-US countries will display a helper text below the tax id input field
   - When this value changes, we display a toast alerting the user that their tax id is being validated
- A new event `tax_id_invalid` is returned and emitted if the tax id is invalid.

## Target release date 🗓️
10/10

## Preview 📷
| Desc  | Screenshot   |
| ------- | ------- |
| Non-US Helper Text | ![Screenshot 2024-05-23 at 12 35 29 PM](https://github.com/linode/manager/assets/125309814/148d6139-32da-4824-a962-8d9b00ae4d9b) |
| Event Notification / Event Page | ![Screenshot 2024-05-23 at 12 12 36 PM](https://github.com/linode/manager/assets/125309814/8a077a0c-0d18-4f17-a7d9-1a94b86fb68f) ![Screenshot 2024-05-23 at 12 12 14 PM](https://github.com/linode/manager/assets/125309814/eccffa46-141a-4885-bb6d-e2b6fbe66d4f) |
| Non-US Helper Text | ![Screenshot 2024-05-24 at 11 06 19 AM](https://github.com/linode/manager/assets/125309814/8eb11c3b-a6ce-43d0-8dc2-346b1dacdd41) |

## How to test 🧪

### Prerequisites
- If you want to test the events part of this, you'll need to setup DevEnv (see me about details)

### Reproduction steps
- Pull down branch
- Turn on feature flag for `TaxId`
- Edit tax id field: http://localhost:3000/account/billing/edit 

### Verification steps
- Observe no helper text is visible and toast doesn't appear for US locations
- For non-US countries, observe both helper text and toast when field if changed

## As an Author I have considered 🤔

*Check all that apply*
